environ.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"os"
+	"strings"
+)
+
+const clientsKey = "XLANG_CLIENTS"
+const paramKey = "XLANG_PARAMS_"
+
+func ReadMatrixFromEnviron() Matrix {
+
+	clients := strings.Split(os.Getenv(clientsKey), ",")
+
+	params := make(Params, 0)
+	paramKeys := make(ParamKeys, 0)
+
+	for _, e := range os.Environ() {
+
+		found := strings.Index(e, paramKey)
```

Comment:
Even better, right now I'm just looking for those keys to exist anywhere in the string, which isn't exactly right.